presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1174,7 +1187,9 @@ public HiveOutputTableHandle beginCreateTable(ConnectorSession session, Connecto
                     partitionUpdate.getUpdateMode(),
                     partitionUpdate.getWritePath(),
                     partitionUpdate.getTargetPath(),
-                    fileNamesForMissingBuckets,
+                    fileNamesForMissingBuckets.stream()
+                            .map(fileName -> new FileWriteInfo(fileName, fileName))
```

Comment:
What about the static factory method we talked about ? :)

Reply:
Given that we used `writeFileName` and `targetFileName`, do we still need a factory method as well as `Optional` field?